tests/test_sh_class.py

```diff
@@ -0,0 +1,137 @@
+"""
+Tests for spherical harmonic class
+"""
+import pytest
+import numpy as np
+import pyfar as pf
+from spharpy.spherical import SphericalHarmonics
+from spharpy.samplings import gaussian, calculate_sampling_weights, equiangular
+
+def test_sphharm_init():
+    coordinates = equiangular(n_points=8)
+    sph_harm = SphericalHarmonics(n_max=2, coordinates=coordinates)
+    assert sph_harm.n_max == 2
+    assert np.all(sph_harm.coordinates == coordinates)
+    assert sph_harm.basis_type == 'real'
+    assert sph_harm.normalization == 'n3d'
+    assert sph_harm.channel_convention == 'acn'
+    assert sph_harm.inverse_method == 'pseudo_inverse'
+    assert sph_harm.condon_shortley == 'auto'
+
+def test_sphharm_init_invalid_coordinates():
+    with pytest.raises(TypeError, 
+                       match="coordinates must be a pyfar.Coordinates " \
+                       "object or spharpy.SamplingSphere object"):
+        SphericalHarmonics(n_max=2, coordinates=[0, 0, 1])
+
+def test_sphharm_init_invalid_n_max():
+    coordinates = pf.Coordinates(1, 0, 0)
+    with pytest.raises(ValueError):
+        SphericalHarmonics(n_max=-1, coordinates=coordinates)
+
+def test_sphharm_compute_basis():
+    coordinates = gaussian(n_points=8)
+    sph_harm = SphericalHarmonics(n_max=2, coordinates=coordinates)
+    assert sph_harm.basis is not None
+
+def test_sphharm_compute_basis_gradient():
+    coordinates = equiangular(n_points=8)
+    sph_harm = SphericalHarmonics(n_max=2, coordinates=coordinates)
+    assert sph_harm.basis_gradient_theta is not None
+    assert sph_harm.basis_gradient_phi is not None
+    
+def test_sphharm_compute_inverse_quad():
+    coordinates = gaussian(n_points=4)
+    weights = calculate_sampling_weights(coordinates)
+    coordinates.weights = weights
+    sh = SphericalHarmonics(2, coordinates, inverse_method = 'quadrature')
+    assert sh.basis_inv is not None
+
+def test_sphharm_compute_inverse_pseudo_inv():
+    coordinates = gaussian(n_points= 5)
+    sh = SphericalHarmonics(2, coordinates, 
+                            inverse_method = 'pseudo_inverse')
+    assert sh.basis_inv is not None
+
+def test_compute_basis_caching():
+    n_max = 2
+    points = np.random.randint(4, 10)
```

Comment:
```suggestion
    rng = np.random.default_rng()
    points = rng.integers(4, 10)
```

this is giving a ruff warning, since it is deprecated in numpy